### PR TITLE
Support selecting derived biotypes

### PR DESCRIPTION
**Pull request by GavinHuttley**
## Summary by Sourcery

Add support for selecting derived biotypes like gene, transcript, CDS, and mRNA, with improved flexibility in feature retrieval

New Features:
- Introduce ability to query derived biotypes (gene, transcript, CDS, mRNA) in addition to standard Ensembl biotypes
- Add support for canonical and non-canonical feature selection

Enhancements:
- Refactor feature retrieval methods to support more flexible querying
- Improve type handling and data conversion for different feature types
- Add mapping between derived biotypes and Ensembl biotypes

Tests:
- Add comprehensive tests for retrieving different biotype features
- Verify canonical and non-canonical feature retrieval
- Test different derived biotype scenarios

### CODE REVIEW

File: tests/test_annotation.py [hunk 1]
```diff
@@ -112,9 +112,53 @@
...
+            ),
+        ),
+    )
     assert cds.stable_id == "F53H8.4.1"  # ID from ensembl.org
+    assert cds.spans.shape[1] == 2
+    assert cds.spans.shape[0] > 1
+
+
+@pytest.mark.parametrize("biotype", ["mrna", "transcript"])
+def test_canonical_transcript(genome_dir, biotype):
```
[the remark this answers is not in the document]
Comment (GavinHuttley) [reply tests/test_annotation.py:130]: not sure if such edge cases can even exist!
